### PR TITLE
[New] `style-prop-object`: Add `allow` option

### PR DESCRIPTION
**Pull request by hornta**
Fixes https://github.com/yannickcr/eslint-plugin-react/issues/1813

### CODE REVIEW

**Comment by jseminck** [comment]
I think that we also make sure that `React.createElement(MyCustomElem, { style: "myStyle" })` works if "MyCustomElem" is an exception?

**Comment by hornta** [comment]
@jseminck Oh yes of course. Thank you. I'll try to cover that aswell.

**Comment by hornta** [comment]
@jseminck I've updated this PR with `React.createElement`. 
I've also managed to get 100% coverage in this rule 👍

**Comment by jseminck** [comment]
Cool. LGTM from my side, as I attempted to add this improvement myself then my solution was very similar. 😄

I haven't been involved in this project for a while though so it will need a review from other people as well.

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,6 @@ sftp-config.json
 # Only apps should have lockfiles
 yarn.lock
 package-lock.json
+
+# Visual Code folder
+.vscode
```
Comment (ljharb) [inline .gitignore]: please revert this; specific IDE files should go in your personal global gitignore, and not clutter every repo you use.

File: lib/rules/style-prop-object.js [hunk 2]
```diff
@@ -53,6 +67,16 @@ module.exports = {
           && node.callee.property.name === 'createElement'
           && node.arguments.length > 1
         ) {
+          if (node.arguments[0].name) {
+            // store name of component
+            const componentName = node.arguments[0].name;
+
+            // allowed list contains the name
+            if (allowed.indexOf(componentName) > -1) {
```
Comment (ljharb) [inline lib/rules/style-prop-object.js]: we should be able to make a `Set` on line 38, so this doesn't need to be O(n)

File: lib/rules/style-prop-object.js [hunk 3]
```diff
@@ -71,6 +95,21 @@ module.exports = {
           return;
         }
 
+        // parent element is a JSXOpeningElement
+        if (node.parent && node.parent.type === 'JSXOpeningElement') {
+          // store parent element
+          const jsxOpeningElement = node.parent;
```
Comment (ljharb) [inline lib/rules/style-prop-object.js]: this can be stored prior to the "if", so we don't have to extract it multiple times.

**Comment by ljharb** [comment]
@hornta would you mind checking the "allow edits" checkbox on the right hand side of the PR?

**Comment by hornta** [comment]
> @hornta would you mind checking the "allow edits" checkbox on the right hand side of the PR?

Done. I totally forgot this one! :fearful:

**Comment by evosch** [comment]
Sorry for the duplicate pull request, I'm a bit new to this. How should I get the changes in? Do a PR on [hornta/eslint-plugin-react](https://github.com/hornta/eslint-plugin-react)?

**Comment by ljharb** [comment]
@evosch yep! Then you can post the link here, and i can merge it in to the PR if needed.

**Comment by ljharb** [comment]
Just noticed https://github.com/hornta/eslint-plugin-react/pull/1 was filed. I'll pull that in soon.

**Comment by koba04** [comment]
Are there any blockers to be merged this?

**Comment by ron23** [comment]
Hey, @hornta I too would love this to be merged. Any plans moving forward with this?

**Pull request merged**